sktime/transformations/series/compose.py

```diff
@@ -91,8 +124,239 @@ def transform(self, Z, X=None):
+            If it is None all columne are transformed.
+
+    Attributes
+    ----------
+    transformers_ : dict of {str : transformer}
+        Maps columns to transformers
+    columns_ : list of str
+        Names of columns that are supposed to be transformed.
+
+
+    Example
+    -------
+    >>> from sktime.datasets import load_longley
+    >>> from sktime.transformations.series.detrend import Detrender
+    >>> from sktime.transformations.series.compose import ColumnwiseTransformer
+
+    >>> y, X = load_longley()
+    >>> transformer = ColumnwiseTransformer(Detrender())
+    >>> yt = transformer.fit_transform(X)
+    """
+
+    _required_parameters = ["transformer"]
+
+    def __init__(self, transformer, columns=None):
+        self.transformer = transformer
+        self.columns = columns
+        super(ColumnwiseTransformer, self).__init__()
+
+    def fit(self, Z, X=None):
+        """
+        Fit data.
+
+        Iterates over columns (series) and applies
+        the fit function of the transformer.
+
+        Parameters
+        ----------
+        Z : pd.Series, pd.DataFrame
+
+        Returns
+        -------
+        self : an instance of self
+        """
+        self._is_fitted = False
+
+        z = check_series(Z, allow_numpy=False)
+
+        # cast to pd.DataFrame in univariate case
+        if isinstance(z, pd.Series):
+            z = z.to_frame()
+
+        # check that columns are None or list of strings
+        if self.columns is not None:
+            if not isinstance(self.columns, list) and all(
+                isinstance(s, str) for s in self.columns
+            ):
+                raise ValueError("Columns need to be a list of strings or None.")
+
+        # set self.columns_ to columns that are going to be transformed
+        # (all if self.columns is None)
+        self.columns_ = self.columns
+        if self.columns_ is None:
+            self.columns_ = z.columns
+
+        # make sure z contains all columns that the user wants to transform
+        self._check_columns(z)
+
+        # fit by iterating over columns
+        self.transformers_ = {}
+        for colname in self.columns_:
+            transformer = clone(self.transformer)
+            self.transformers_[colname] = transformer
+            self.transformers_[colname].fit(z[colname], X)
+        self._is_fitted = True
+        return self
+
+    def transform(self, Z, X=None):
+        """Transform data.
+
+        Returns a transformed version of Z by iterating over specified
+        columns and applying the univariate series transformer to them.
+
+        Parameters
+        ----------
+        Z : pd.Series, pd.DataFrame
+
+        Returns
+        -------
+        Z : pd.Series, pd.DataFrame
+            Transformed time series(es).
+        """
+        self.check_is_fitted()
+        z = check_series(Z)
+
+        # transform to pd.Dataframe in univariate case
+        is_series = False
```

Comment:
there's some code duplication here, lines 253-257 are used below again, right?

Reply:
Thanks for your feedback @mloning! 

If I wrap this into a function it would look smth like this: 
``` python
def check_is_series(z): 
    name_of_z = none
    if isinstance(z, pd.Series):
            is_series = True
            name_of_z = z.name
            z = z.to_frame() 
    return z, name_of_z
```      
and then above we would have 
                `z, name_of_z = check_is_series(z)`
even though we only need `name_of_z` if the user passes a pd.series. So I'm not sure whether wrapping this into a function makes it less readable? But I'm happy to change it.

Reply:
Thanks @mloning, I moved it into a wrapper function now :)